### ISSUE TITLE
210527 인공지능 오피스 아워 이슈 : 모델 학습 코드 리뷰

### ISSUE THREAD

**Issue by makeitmin** [mirror-look/mirror-look-origin#55]
In GitLab by @Jaemin on May 27, 2021, 17:49

## 개발 시 어려운 점을 아래 리스트 중에 체크해주세요.
| 이슈 Label| 해당 사항 체크(✅,❎) |
| ------ | ------ |
| 전체적인 아이디어, 프로젝트 진행 방향성에 관한 이슈 |  |
| To-do 기능 구현의 어려움 |  |
| 딥러닝 결과 해석의 어려움 |  |
| 버그 |  |
| 서버 이슈 |  |
| 속도 이슈 | ✅ |
| gitlab 이슈 |  |
| 그 외 | ✅ |

- 이슈에 대한 자세한 설명 :

### 1. 개발 시 이슈가 무엇이었는지 예상 동작과 현재 동작에 대해 자세히 기입해주세요.

- 어제 피드백 해주셨던 object detection model을 이용해서 Dress 카테고리의 데이터 수를 undersampling을 하려고 했지만, 제가 예상 시간 계산해본 결과..13시간(?)이 걸려서 포기했습니다...ㅠ
- 그리고 어제 모델을 학습 시키고 잤는데... 이전에 한번도 발생한 적 없었던 메모리 문제 때문에 kernel이 중단되서 학습이 멈췄습니다.

### 4. 이슈를 어떻게 고칠 것인지 자세히 설명해주세요. (해결 방법을 못 찾을 시, 5. 궁금한 점에 기입해주세요)

- 어제 발생했던 메모리 문제로 인한 학습 중단에 대해서 코치님께 코드 리뷰를 받아보고, 피드백을 반영하고 다시 학습을 시켜본다.

**Comment by makeitmin** [comment]
In GitLab by @Jaemin on May 27, 2021, 19:02

changed title from **210527 인공지능 오피스 아워 이슈 : 모델 학습 코드 {-피드백-}** to **210527 인공지능 오피스 아워 이슈 : 모델 학습 코드 {+리뷰+}**

**Comment by makeitmin** [comment]
In GitLab by @hyoje on May 27, 2021, 21:39

1. Global Average Pooling

vgg = tf.keras.applications.VGG16(
    include_top=False, weights='imagenet', input_tensor=None,
    input_shape=None, pooling='avg')

model = tf.keras.Sequential([
    vgg,
    Dense(10),
]
)

이런식으로 쓰면 저는 에러가 안나던데 확인해보셔요

그리고 제가 기억을 못 할 수도 있으니 추가 코멘트 필요한 것 있으시면 코멘트나 이슈 남겨주셔요.
